src/backend.rs

```diff
@@ -60,15 +68,23 @@ impl Backend {
             network: Default::default(),
         };
 
-        self.veilid_api = Some(api_startup_config(update_callback, config_inner).await.map_err(|e| anyhow!("Failed to initialize Veilid API: {}", e))?);
+        if self.veilid_api.is_none() {
+            self.veilid_api = Some(Arc::new(api_startup_config(update_callback, config_inner).await.map_err(|e| anyhow!("Failed to initialize Veilid API: {}", e))?));
+        }
 
         Ok(())
     }
 
     pub async fn stop(&self) -> Result<()> {
         println!("Stopping Backend...");
         if let Some(veilid) = &self.veilid_api {
-            veilid.clone().shutdown().await;
+            if let Ok(veilid) = Arc::try_unwrap(Arc::clone(veilid)) {
```

Comment:
What is the purpose of doing a clone? Can we not shutdown the instance held in the arc?